math/FunMath.cpp

```diff
@@ -1 +1,21 @@
 #include "FunMath.h"
+
+/**
+    Adds all element from a that do not match with b
```

Comment:
This comment is confusing—are we removing elements from a that **are** in b or that **aren't** in b?

You are missing comments about the type and nature of parameters and return values.